pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -671,7 +671,8 @@ public CompletableFuture<Boolean> hasSchema() {
         return brokerService.pulsar().getSchemaRegistryService().getSchema(getSchemaId()).thenApply(Objects::nonNull)
                 .exceptionally(e -> {
                     Throwable ex = e.getCause();
-                    if (ex instanceof SchemaException || !((SchemaException) ex).isRecoverable()) {
+                    if (brokerService.pulsar().getConfig().isSchemaLedgerForceRecovery()
+                            && !((SchemaException) ex).isRecoverable()) {
```

Comment:
What if `ex` isn't `SchemaException` instance? Please also handle that case.

Reply:
yes, fixed it.